src/test/rgw/test_multi.py

```diff
@@ -499,39 +506,49 @@ def gen_bucket_name():
     return run_prefix + '-' + str(num_buckets)
 
 class RGWMulti:
-    def __init__(self, num_clusters, gateways_per_cluster):
+    def __init__(self, zg_num, num_clusters, gateways_per_cluster, base_port, base_port_master_zg):
         self.num_clusters = num_clusters
 
-        self.base_port = 8000
+        self.zg_num = zg_num
+        self.zg_name = 'zg' + str(zg_num)
+
+        self.base_port = base_port
+        self.base_port_master_zg = base_port_master_zg
 
         self.clusters = {}
         for i in range(num_clusters):
-            self.clusters[i] = RGWCluster(i + 1, self.base_port + i * gateways_per_cluster, gateways_per_cluster)
+            self.clusters[i] = RGWCluster(self.zg_num, i + 1, 'zg' + str(zg_num) + '-c' + str(i + 1), self.base_port + i * gateways_per_cluster, gateways_per_cluster)
 
     def setup(self, bootstrap, tenant):
-        global realm
+        global master_zg
         global realm_credentials
         global user
 
-        realm_credentials = RGWRealmCredentials(gen_access_key(), gen_secret())
-        realm = RGWRealm('earth', realm_credentials, self.clusters)
+        is_master_zg = (self.base_port == self.base_port_master_zg)
+        if is_master_zg:
+            realm_credentials = RGWRealmCredentials(gen_access_key(), gen_secret())
+        rgw_zg = RGWZonegroup('earth', realm_credentials, self.clusters)
+        
+        if is_master_zg:
+            master_zg = rgw_zg
 
         if bootstrap:
             log(1, 'bootstrapping clusters')
             self.clusters[0].start()
-            realm.init_zone(self.clusters[0], 'us', 'us-1', self.base_port)
+            rgw_zg.init_zone(self.clusters[0], self.zg_name, self.zg_name + '-1', self.base_port, self.base_port_master_zg)
 
             for i in range(1, self.num_clusters):
                 self.clusters[i].start()
-                realm.init_zone(self.clusters[i], 'us', 'us-' + str(i + 1), self.base_port)
+                rgw_zg.init_zone(self.clusters[i], self.zg_name, self.zg_name + '-' + str(i + 1), self.base_port, self.base_port_master_zg)
         else:
             for i in range(0, self.num_clusters):
-                realm.add_zone(self.clusters[i], 'us', 'us-' + str(i + 1), (i == 0))
+                rgw_zg.add_zone(self.clusters[i], self.zg_name, self.zg_name + '-' + str(i + 1), (i == 0))
 
-        realm.meta_checkpoint()
+        rgw_zg.meta_checkpoint()
```

Comment:
i think we still want a `RGWRealm` in some form. for `meta_checkpoint()` in particular, we want to wait for all zones in all zonegroups to catch up to the metadata master zone